### PR TITLE
created validToken()

### PR DESCRIPTION
**Pull request by tomkettle1**
This should work fine, but I can't test until I get docker set up w/ the database or whatever so @TimMcGilly would you mind checking this runs

### CODE REVIEW

**Comment by tomkettle1** [comment]
(Or @dislocated-su if you've got postgresql working on your machine)

**Comment by TimMcGilly** [comment]
Will test in moment.

File: index.js [hunk 1]
```diff
@@ -42,6 +42,15 @@ function servePage(path, res) {
     });
 }
 
+function validToken(token){
```
Comment (TimMcGilly) [inline index.js:45]:
```suggestion
async function validToken(token){
```
Function needs to be async to have await in it.

File: index.js [hunk 2]
```diff
@@ -42,6 +42,15 @@ function servePage(path, res) {
...
+    const result = await client.query({
+        text: 'SELECT teamid FROM team WHERE teamtoken = $1',
+        values: [token],
+    });
```
Comment (TimMcGilly) [inline index.js:49]: Possible needs error handling?

File: index.js [hunk 3]
```diff
@@ -42,6 +42,15 @@ function servePage(path, res) {
...
+    if (result.rowCount === 0) {return false;}
+    else {return true;}
```
Comment (TimMcGilly) [inline index.js:51]:
```suggestion
    return !result.rowCount === 0;
```

**Review by TimMcGilly — CHANGES_REQUESTED** [review]

**Comment by TimMcGilly** [comment]
This also break line 103 `const teamID = result.rows[0].teamid;` which gets team id but depends on query. Not sure best way to fix. Maybe change function to return teamID and if invalid it returns null. @dislocated-su Any thoughts?

**Comment by tomkettle1** [comment]
ok, fixed everything except adding error catching @TimMcGilly

File: index.js [hunk 4]
```diff
@@ -42,6 +42,18 @@ function servePage(path, res) {
     });
 }
 
+async function validToken(token){
```
Comment (dislocated-su) [inline index.js:45]: Function returns a teamid, not a boolean.
Name needs to reflect the fact that it returns a teamid.

**Review by dislocated-su — CHANGES_REQUESTED** [review]
Almost done, needs some minor changes.

File: index.js [hunk 5]
```diff
@@ -42,6 +42,18 @@ function servePage(path, res) {
...
+    const result = await client.query({
+        text: 'SELECT teamid FROM team WHERE teamtoken = $1',
+        values: [token],
+    });
+    if (!result.rowCount === 0) {
+        return result[0].token;
+    }
+    else {return null;}
```
Comment (TimMcGilly) [inline index.js:53]:
```suggestion
return null;
```
Doesn't need else.

**Comment by TimMcGilly** [comment]
Will test on my side later once all changes including Andrey's are done.

**Comment by TimMcGilly** [comment]
Are we not using refactoring this anymore as it seemed like a good idea? @tomkettle1

**Pull request closed without merging**